File: 2859-sum-of-values-at-indices-with-k-set-bits/2859-sum-of-values-at-indices-with-k-set-bits.py
class Solution:
    def sumIndicesWithKSetBits(self, nums: List[int], k: int) -> int:

        count = 0
        while num:
            count += num & 1
            num >>= 1
        return count

    def sumIndicesWithKSetBits(self, nums: list[int], k: int) -> int:
        total_sum = 0
        for i, num in enumerate(nums):
            count_bits = 0
            index = i
            while index:
                count_bits += index & 1
                index >>= 1
            if count_bits == k:
                total_sum += num
        return total_sum

File: 2859-sum-of-values-at-indices-with-k-set-bits/2859-sum-of-values-at-indices-with-k-set-bits.java
import java.util.List;

public class Solution {
    
    private int countSetBits(int num) {
        int count = 0;
        while (num > 0) {
            count += num & 1;
            num >>= 1;
        }
        return count;
    }

    public int sumIndicesWithKSetBits(List<Integer> nums, int k) {
        int totalSum = 0;
        for (int i = 0; i < nums.size(); i++) {
            int countBits = countSetBits(i);
            if (countBits == k) {
                totalSum += nums.get(i);
            }
        }
        return totalSum;
    }
}
